docs/guide/pages.md

```diff
@@ -0,0 +1,239 @@
+        type: 'area',
+        options: {
+          widgets: {
+            '@apostrophecms/rich-text': {},
+            '@apostrophecms/image': {}
+          }
+        }
+      }
+    },
+    group: {
+      basics: {
+        fields: ['title', 'subtitle', 'main']
+      }
+    }
+  }
+};
+```
+
+See the [content schema](/guide/introduction.md#content-schemas) page for more on configuring fields.
+
+## Page template essentials
+
+You will need to add a template file for each page type. The only exception to that rule is if your page type module extends another page type that already has a template.
+
+Page templates are added in a `views` directory in the module as `page.html`. The template for the previous example's default page would be `modules/default-page/views/page.html`. A very simple page template for the Default page might look like this:
+
+```django
+{# modules/default-page/views/page.html #}
+{% extends "layout.html" %}
+
+{% block main %}
+  <header>
+    <h1>{{ data.page.title }}</h1>
+    {% if data.page.subtitle %}
+      <p>{{ data.page.subtitle }}</p>
+    {% endif %}
+  </header>
+  {% area data.page, 'main' %}
+{% endblock %}
+```
+
+There are a number of things at work here.
+
+### The template is extending a `layout.html` template
+
+```django
+{% extends "layout.html" %}
+```
+
+`layout.html` is a base level template [used in official Apostrophe boilerplates](https://github.com/apostrophecms/a3-boilerplate/blob/main/views/layout.html) and placed in `views/layout.html`. It is used to add markup for things that belong on every page, such as the website navigation and footer. It extends the `outerLayout.html` template from Apostrophe core, but provides a layer to customize the page wrapper while not overwriting `outerLayout.html`.
+
+The layout template might look something like this:
+
+```django
+{% extends data.outerLayout %}
+
+{% block beforeMain %}
+<div>
+  <header>
+    {# Page header code: logo, navigation, etc. #}
+  </header>
+  <main>
+{% endblock %}
+
+{% block main %}
+  {# Page body content. Pages templates normally override this. #}
+{% endblock %}
+
+{% block afterMain %}
+  </main>
+  <footer>
+    {# Page header code: contact information, secondary navigation, etc. #}
+  </footer>
+</div>
+{% endblock %}
+```
+
+### We are inserting page template markup in a template block
+
+```django
+{% block main %}
+{% endblock %}
+```
+
+Apostrophe uses the Nunjucks template language, which has a [block system](https://mozilla.github.io/nunjucks/templating.html#block) for injecting markup into lower-level templates. The block system involves placing a `block` tag in the root-level template file, then using those blocks in higher-level templates to insert markup.
+
+In the Default page example, the layout template's `beforeMain` and `afterMain` blocks are not replaced, so rendered Default pages use that markup as-is. The Default page template does use the `main` block, so the layout template's version is replaced.
```

Comment:
```suggestion
```
I almost dont think we need this. I felt like the example did a fairly good job of illustrating everything, but when I got here I found myself tripping over this paragraph, and in the end realizing it said what I already perceived to be true.

Reply:
"Since we used the main block here it will overwrite the layout template main block."